pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/NullAssignment.xml

```diff
@@ -129,4 +149,41 @@ public class Foo {
 }
      ]]></code>
     </test-code>
+
+    <test-code>
+        <description>[java] NullAssignment false positive - initialization #629</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class NullAssignmentFP {
+    public void foo() {
+        Type result = someHash.computeIfAbsent(a, _unused -> test ? truthy : null);
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>[java] NullAssignment false positive - actual assignment #629</description>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+public class NullAssignmentFP {
+    public void foo() {
+        Type result;
+        result = someHash.computeIfAbsent(a, _unused -> test ? truthy : null);
```

Comment:
I'm not sure this is actually right... the lambda is a find boundary. From it's point of view, it's a `return test ? truthy : null` which would be ok (it's not an assignment). For the assignmet you would have to understand the semantics of `computeIfAbsent`

Reply:
You're right - in this specific example (for computeIfAbsent) the result is a null assignment. But it won't be always the "computeIfAbsent" usage case....

I'll change the test to not expect a violation here, otherwise we'll probably get many false positives.

Reply:
@jsotuyod I've adjusted the test+implementation to ignore lambdas.
I've also rebase this PR against master, it should be ready now.